File: problems/Easy/997. Find the Town Judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        if(n==1){
            return 1;
        }
        map<int,int> trusts, trusted;
        for(auto i:trust){
            trusts[i[0]]++;
            trusted[i[1]]++;
        }
        for(auto i:trusted){
            if(i.second==n-1 && trusts[i.first]==0){
                return i.first;
            }
        }
        return -1;
    }
};

File: problems/Easy/997. Find the Town Judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        mp = [0]*(n+1)
        for i in trust:
            mp[i[1]] = mp[i[1]] + 1
            mp[i[0]] = mp[i[0]] - 1
        # print(mp)
        for i in range(1, n+1):
            if mp[i] == n-1:
                return i
        return -1